src/Compilers/CSharp/Test/Symbol/DocumentationComments/DocumentationCommentCompilerTests.cs

```diff
@@ -3782,6 +3782,167 @@ Part 2.
             Assert.Equal(string.Format(expectedTemplate, TestHelpers.AsXmlCommentText(xmlFilePath)), actual);
         }
 
+        [Fact]
+        [WorkItem(21348, "https://github.com/dotnet/roslyn/issues/21348")]
+        public void IncludedName_TypeParamAndTypeParamRefHandling()
+        {
+            var xml = @"
+<root>
+    <target>
+    Included section
+    <summary>
+      See <typeparam/>.
+      See <typeparam name=""X""/>.
+      See <typeparam name=""Y""/>.
+      See <typeparam name=""XY""/>.
+    </summary>
+    <remarks></remarks>
+    </target>
+    <target>
+    Included section
+    <summary>
+      See <typeparamref/>.
+      See <typeparamref name=""X""/>.
+      See <typeparamref name=""Y""/>.
+      See <typeparamref name=""XY""/>.
+    </summary>
+    <remarks></remarks>
+    </target>
+</root>
+";
+
+            var xmlFile = Temp.CreateFile(extension: ".xml").WriteAllText(xml);
+            var xmlFilePath = xmlFile.Path;
+
+            var includeElementTemplate = @"<include file='{0}' path='//target'/>";
+            string includeElement = string.Format(includeElementTemplate, xmlFilePath);
+
+            var sourceTemplate = @"
+
+/// {0}
+class OuterClass<X>
+{{
+    /// {0}
+    class InnerClass<Y>
```

Comment:
Is the binding for a method significantly enough different that we should add a test for that?

Reply:
It's not, but this test covers both the outer/inner classes, and a method inside